subxt/src/metadata/metadata_type.rs

```diff
@@ -485,11 +543,60 @@ impl TryFrom<RuntimeMetadataPrefixed> for Metadata {
             .find(|ty| ty.ty.path.segments == ["sp_runtime", "DispatchError"])
             .map(|ty| ty.id);
 
+        let extrinsic_ty = metadata
+            .types
+            .resolve(metadata.extrinsic.ty.id)
+            .ok_or(InvalidMetadataError::MissingType(metadata.extrinsic.ty.id))?;
+
+        let Some(call_id) = extrinsic_ty.type_params
+            .iter()
+            .find(|ty| ty.name == "Call")
+            .and_then(|ty| ty.ty)
+            .map(|ty| ty.id) else {
+            return Err(InvalidMetadataError::MissingCallType);
+        };
+
+        let type_def_variant = get_type_def_variant(call_id)?;
+
+        let mut extrinsics = HashMap::<(u8, u8), ExtrinsicMetadata>::new();
+        for variant in &type_def_variant.variants {
+            let pallet_name: Arc<str> = variant.name.to_string().into();
+            let pallet_index = variant.index;
+
+            // Pallet variants must contain one single call variant.
+            // In the following form:
+            //
+            // enum RuntimeCall {
+            //   Pallet(pallet_call)
+            // }
```

Comment:
I think it's optional that a pallet actually has any calls in it; do we _need_ to find them all here?

Ie if a pallet has no calls, then there's no need to try and generate a variant for it at all in the codegen?